src/CommandingOfficers/CommanderAve.java

```diff
@@ -0,0 +1,646 @@
+      disconnected.remove(front);
+    }
+
+    // Any tiles that are no longer being fed snow will melt over time.
+    for(XYCoord dis : disconnected )
+    {
+      int oldVal = snowMap[dis.xCoord][dis.yCoord];
+      snowMap[dis.xCoord][dis.yCoord] = (oldVal - SNOW_MELT_RATE < 0)? 0 : oldVal - SNOW_MELT_RATE;
+      if( snowLoggingEnabled ) log("Snow at " + dis + " melting from " + oldVal + " to " + snowMap[dis.xCoord][dis.yCoord]);
+    }
+
+    // Update weather forecast.
+    ArrayList<MapChangeEvent.EnvironmentAssignment> tiles = new ArrayList<MapChangeEvent.EnvironmentAssignment>();
+    for (XYCoord coord : toSnow)
+    {
+      Environment newEnvi = Environment.getTile(gameMap.getEnvironment(coord).terrainType, Weathers.SNOW);
+      int duration = snowMap[coord.xCoord][coord.yCoord] / SNOW_MELT_RATE;
+      tiles.add(new MapChangeEvent.EnvironmentAssignment(coord, newEnvi, duration));
+    }
+    if( !tiles.isEmpty())
+    {
+      GameEvent event = new MapChangeEvent(tiles);
+      outEvents.add(event);
+    }
+
+    log("---------- FINAL state after relevel -------------------" );
+    log(getSnowMapAsString());
+  }
+
+  public int[][] getSnowMapClone()
+  {
+    int xsize = snowMap.length;
+    int ysize = snowMap[0].length;
+    int[][] snowClone = new int[xsize][ysize];
+    for( int y = 0; y < ysize; ++y )
+    {
+      for( int x = 0; x < xsize; ++x )
+      {
+        snowClone[x][y] = snowMap[x][y];
+      }
+    }
+    return snowClone;
+  }
+
+  public String getSnowMapAsString()
+  {
+    StringBuffer buffer = new StringBuffer();
+    for( int y = 0; y < snowMap[0].length; ++y )
+    {
+      for( int x = 0; x < snowMap.length; ++x )
+      {
+        buffer.append(String.format("%5d", snowMap[x][y])).append(' ');
+      }
+      buffer.append('\n');
+    }
+    return buffer.toString();
+  }
+
+  /**
+   * Nix permanently increases the range of Ave's passive snow effect,
+   * and converts all spaces within that range to snow.
+   */
+  private static class NixAbility extends CommanderAbility
+  {
+    private static final String NIX_NAME = "Nix";
+    private static final int NIX_COST = 1;
+    private static final int NIX_BUFF = 10; // Standard 10
+
+    private int numActivations = 0;
+
+    CommanderAve Ave;
+    COModifier damageMod = null;
+
+    NixAbility(CommanderAve commander)
+    {
+      super(commander, NIX_NAME, NIX_COST);
+      Ave = commander;
+      damageMod = new CODamageModifier(NIX_BUFF);
+      AIFlags = PHASE_TURN_START | PHASE_TURN_END;
+    }
+
+    @Override
+    protected void adjustCost()
+    {
+      // Override default cost-increase behavior to make this get more expensive faster.
+      numActivations++;
+      myPowerCost = 1 + (numActivations*3);
+    }
+
+    @Override
+    protected void perform(MapMaster gameMap)
+    {
+      // Increase Ave's sphere of influence.
+      Ave.MAX_SNOW_SPREAD_RANGE++;
+
+      // Buff units.
+      damageMod.apply(Ave);
```

Comment:
This line produces an ATK leak.
It might be thematic for Ave to get a firepower bonus in snow (that she can permanently increase), but giving her permanent, stacking global firepower seems a little much.

Reply:
Ahh, good catch. Maybe we should make `apply` private, and make some way to signal if the modifier should be permanent or transient.

Reply:
> Ahh, good catch. Maybe we should make `apply` private, and make some way to signal if the modifier should be permanent or transient.

I don't know that we could make `apply` private without making it impossible for the CO to do it, haha.